android/app/src/main/java/app/organicmaps/editor/OsmLoginFragment.java

```diff
@@ -48,15 +52,41 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
     mLoginInput = view.findViewById(R.id.osm_username);
     mPasswordInput = view.findViewById(R.id.osm_password);
     mLoginButton = view.findViewById(R.id.login);
-    mLoginButton.setOnClickListener((v) -> login());
     mLostPasswordButton = view.findViewById(R.id.lost_password);
-    mLostPasswordButton.setOnClickListener((v) -> Utils.openUrl(requireActivity(), Constants.Url.OSM_RECOVER_PASSWORD));
     Button registerButton = view.findViewById(R.id.register);
     registerButton.setOnClickListener((v) -> Utils.openUrl(requireActivity(), Constants.Url.OSM_REGISTER));
     mProgress = view.findViewById(R.id.osm_login_progress);
     final String dataVersion = DateUtils.getShortDateFormatter().format(Framework.getDataVersion());
     ((TextView) view.findViewById(R.id.osm_presentation))
         .setText(getString(R.string.osm_presentation, dataVersion));
+
+    if (BuildConfig.FLAVOR.equals("google"))
+    {
+      // Hide login and password inputs and Forgot password button
+      UiUtils.hide(view.findViewById(R.id.osm_username_container),
+          view.findViewById(R.id.osm_password_container),
+          mLostPasswordButton);
+
+      mLoginButton.setOnClickListener((v) -> loginWithBrowser());
+    }
+    else
+    {
+      mLoginButton.setOnClickListener((v) -> login());
+      mLostPasswordButton.setOnClickListener((v) -> Utils.openUrl(requireActivity(), Constants.Url.OSM_RECOVER_PASSWORD));
+    }
+
+    readArguments();
+    if (mArgOAuth2Code != null && !mArgOAuth2Code.isEmpty())
```

Comment:
Can storing this variable as a member be avoided?

Reply:
I introduced `String readOAuth2CodeFromArguments()` method to get rid of the field.